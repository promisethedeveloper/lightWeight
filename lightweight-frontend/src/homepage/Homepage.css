span {
	color: #3ccf4e;
}

.homepage-heading {
	color: #461111;
}

.lead {
	color: #b3541e;
}

h2 {
	color: white;
}

h5 {
	color: #040303;
}
