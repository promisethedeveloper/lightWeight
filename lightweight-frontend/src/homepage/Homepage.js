import React, { useContext } from "react";
import { Link } from "react-router-dom";
import "./Homepage.css";
import UserContext from "../auth/UserContext";
import "bootstrap/dist/css/bootstrap.min.css";

/** Homepage of site.
 *
 * Shows welcome message or login/signup buttons.
 *
 * Routed at /
 *
 * Routes -> Homepage
 */

const Homepage = () => {
	const { currentUser } = useContext(UserContext);

	return (
		<div className="Homepage">
			<div className="container text-center">
				<h1 className="mb-4 font-weight-bold">
					<span className="homepage-heading">LIGHT WEIGHT</span>
				</h1>
				<p className="lead">Sore today, stronger tomorrow.</p>
				{currentUser ? (
					<h5>Welcome, {currentUser.username}!</h5>
				) : (
					<p>
						<Link
							className="btn font-weight-bold mr-3"
							to="/login"
							style={{ backgroundColor: "#B3541E", color: "white" }}
						>
							LOG IN
						</Link>
						<Link
							className="btn font-weight-bold"
							to="/signup"
							style={{ backgroundColor: "#B3541E", color: "white" }}
						>
							SIGN UP
						</Link>
					</p>
				)}
			</div>
		</div>
	);
};

export default Homepage;
