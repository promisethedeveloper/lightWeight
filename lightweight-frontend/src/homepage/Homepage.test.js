import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Homepage from "./Homepage";
import UserContext from "../auth/UserContext";

/** Smoke test */
it("should render without crashing", () => {
	const currentUser = { username: "John" };
	render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<Homepage />
			</UserContext.Provider>
		</MemoryRouter>
	);
});

/** Snapshot test */
it("should match snapshots", () => {
	const currentUser = { username: "John" };

	const { asFragment } = render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<Homepage />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("should render welcome message if there is a current user", () => {
	let currentUser = { username: "Promise" };
	const { getByText } = render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<Homepage />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(getByText("Welcome, Promise!")).toBeInTheDocument();
});

it("should render LOG IN and SIGN UP buttons if there is no current user", () => {
	const { getByText } = render(
		<MemoryRouter>
			<UserContext.Provider value={{}}>
				<Homepage />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(getByText("LOG IN")).toBeInTheDocument();
	expect(getByText("SIGN UP")).toBeInTheDocument();
});
