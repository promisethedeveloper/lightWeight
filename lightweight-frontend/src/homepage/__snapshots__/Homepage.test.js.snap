// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        <span
          class="homepage-heading"
        >
          LIGHT WEIGHT
        </span>
      </h1>
      <p
        class="lead"
      >
        Sore today, stronger tomorrow.
      </p>
      <h5>
        Welcome, John!
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
